autoscaling/custom-compute-classes/reservations/deployment-gpu.yaml

```diff
@@ -0,0 +1,56 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START gke_autoscaling_ccc_reservations_deployment_gpu]
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ccc-reservations
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: ccc-reservations
+  template:
+    metadata:
+      labels:
+        app: ccc-reservations
+    spec:
+      nodeSelector:
+        cloud.google.com/compute-class: ccc-reservations
+      containers:
+        - name: ccc-reservations
+          image: us-docker.pkg.dev/google-samples/containers/gke/hello-app:1.0
+          resources:
+            requests:
+              cpu: 500m
+              memory: 500Mi
```

Comment:
Nitpick (feel free to ignore): Consider reducing the resource request.
[This existing Deployment](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/5be2e5137f44bd13a9cdd003735bf31453c3a47e/quickstarts/hello-app/manifests/helloweb-deployment.yaml#L42) uses `cpu: 200m`.